Sources/MongoSwift/ConnectionString.swift

```diff
@@ -595,7 +595,7 @@ internal class ConnectionString {
             return nil
         }
         let str = String(cString: mechanism)
-        return MongoCredential.Mechanism(str)
+        return try? MongoCredential.Mechanism(str)
```

Comment:
Constructing a `Mechanism` now possibly throws if the name is invalid, but that shouldn't be an issue here because we're getting the mechanism from a `mongoc_uri`. (We'll also be moving away from this type anyway.)